### PR TITLE
Add mongodb workspace

### PR DESCRIPTION
**Pull request by hamza0867**

### CODE REVIEW

File: mongodb/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM gitpod/workspace-full:latest
```
Comment (jankeromnes) [inline mongodb/Dockerfile]: No need to specify `:latest`, it's the default value.

```suggestion
FROM gitpod/workspace-full
```

File: mongodb/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM gitpod/workspace-full:latest
+
+# Install mongodb
+RUN sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 9DA31620334BD75D9DCB49F368818C72E52529D4
+RUN echo "deb [ arch=amd64 ] https://repo.mongodb.org/apt/ubuntu bionic/mongodb-org/4.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-4.0.list
```
Comment (jankeromnes) [inline mongodb/Dockerfile]: I see that your `deb` line mentions `bionic`, however our images are based on `cosmic`. I hope that's not an issue.

File: mongodb/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+# Install mongodb
+RUN sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 9DA31620334BD75D9DCB49F368818C72E52529D4
+RUN echo "deb [ arch=amd64 ] https://repo.mongodb.org/apt/ubuntu bionic/mongodb-org/4.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-4.0.list
+RUN sudo apt-get update \
+ && sudo apt-get install -y mongodb-org \
+ && sudo apt-get clean \
+ && sudo rm -rf /var/cache/apt/* /var/lib/apt/lists/* /tmp/*
+RUN sudo mkdir -p /data/db
+RUN sudo chown gitpod:gitpod -R /data/db
```
Comment (jankeromnes) [inline mongodb/Dockerfile]: Note: Every `RUN` command will create a new Docker layer. I don't think we need to create so many layers here.

Could you please briefly rework these lines from:

```Dockerfile
RUN sudo mkdir -p /data/db
RUN sudo chown gitpod:gitpod -R /data/db
```

to:

```Dockerfile
RUN sudo mkdir -p /data/db \
 && sudo chown gitpod:gitpod -R /data/db
```

please?

File: mongodb/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
+FROM gitpod/workspace-full:latest
+
+# Install mongodb
+RUN sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 9DA31620334BD75D9DCB49F368818C72E52529D4
+RUN echo "deb [ arch=amd64 ] https://repo.mongodb.org/apt/ubuntu bionic/mongodb-org/4.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-4.0.list
+RUN sudo apt-get update \
+ && sudo apt-get install -y mongodb-org \
+ && sudo apt-get clean \
+ && sudo rm -rf /var/cache/apt/* /var/lib/apt/lists/* /tmp/*
```
Comment (jankeromnes) [inline mongodb/Dockerfile]: I think you can replace `sudo apt-get clean` and all these `rm -rf`s with just:

```bash
sudo rm -rf /var/lib/apt/lists/*
```

The others should already be cleaned up automatically by `apt-get`.

File: mongodb/Dockerfile (hunk 3 above)
Comment (hamza0867) [reply mongodb/Dockerfile]: Done

File: mongodb/Dockerfile (hunk 4 above)
Comment (hamza0867) [reply mongodb/Dockerfile]: Done

**Comment by hamza0867** [comment]
I tried to follow your suggestions. If you can please review them it would be great ^_^

**Comment by jankeromnes** [comment]
> I tried to follow your suggestions. If you can please review them it would be great ^_^

Thanks a lot! I'll take a look. 🙂

**Comment by hamza0867** [comment]
It wsa  my first PR ever XD

**Comment by jankeromnes** [comment]
That is amazing. Really well done, and again thank you so much! 👍

**Comment by jankeromnes** [comment]
Update: Hooray, the build passed and we now have a new official image: [gitpod/workspace-mongodb](https://hub.docker.com/r/gitpod/workspace-mongodb)! 🎉

**Pull request merged**